[macOS] Update ScreenCaptureKit API used

eric-carlson opened the pull request:
#### 0b941e1c1cc820380a9560f39012804ead3ea994
<pre>
[macOS] Update ScreenCaptureKit API used
<a href="https://bugs.webkit.org/show_bug.cgi?id=257816">https://bugs.webkit.org/show_bug.cgi?id=257816</a>
rdar://110408877

Reviewed by Youenn Fablet.

Update to the most recent version of ScreenCaptureKit API.

* Source/WebCore/PAL/pal/spi/mac/ScreenCaptureKitSPI.h:
* Source/WebCore/platform/mediastream/mac/ScreenCaptureKitSharingSessionManager.mm:
(-[WebDisplayMediaPromptHelper initWithCallback:]):
(-[WebDisplayMediaPromptHelper contentSharingPicker:didCancelForStream:]):
(-[WebDisplayMediaPromptHelper startObservingPicker:]):
(-[WebDisplayMediaPromptHelper stopObservingPicker:]):
(WebCore::ScreenCaptureKitSharingSessionManager::~ScreenCaptureKitSharingSessionManager):
(WebCore::ScreenCaptureKitSharingSessionManager::cancelPicking):
(WebCore::ScreenCaptureKitSharingSessionManager::promptWithSCContentSharingPicker):
(-[WebDisplayMediaPromptHelper contentSharingPickerDidCancel:forStream:]): Deleted.

Canonical link: <a href="https://commits.webkit.org/264988@main">https://commits.webkit.org/264988@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/21969630d205136ff67c469b2b4a931e691f40b8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/9319 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/9598 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/9826 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/10981 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/9198 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/9328 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/11584 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/9559 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/12085 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/9468 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/10389 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/8034 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/11140 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/7685 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/8498 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/15953 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/8783 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/8649 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/11998 "6 api tests failed or timed out") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/9146 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/7449 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/8360 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2269 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/12584 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/8909 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebCore/platform/mediastream/mac/ScreenCaptureKitSharingSessionManager.mm:
```diff
@@ -168,6 +158,23 @@ - (void)contentSharingPicker:(SCContentSharingPicker *)picker didUpdateWithFilte
+- (void)stopObservingPicker:(SCContentSharingPicker *)picker
+{
+    if (!_observingPicker)
+        return;
```
Should we set _observingPicker to NO?

eric-carlson replied:
We should!

youennf commented on Source/WebCore/PAL/pal/spi/mac/ScreenCaptureKitSPI.h:
```diff
@@ -109,18 +115,18 @@ typedef NS_OPTIONS(NSUInteger, SCContentSharingAllowedPickerModes) {
 };
 
-@interface SCContentSharingPickerConfiguration : NSObject
+@interface SCContentSharingPickerConfiguration (Staging_110281298)
```
Not sure what this change means.

eric-carlson replied:
It is declaring the interfaces for both internal and open source builds, so we don't have to coordinate updating all of the bots to the new SDK.

eric-carlson committed:
[macOS] Update ScreenCaptureKit API used
https://bugs.webkit.org/show_bug.cgi?id=257816
rdar://110408877

Reviewed by Youenn Fablet.

Update to the most recent version of ScreenCaptureKit API.

* Source/WebCore/PAL/pal/spi/mac/ScreenCaptureKitSPI.h:
* Source/WebCore/platform/mediastream/mac/ScreenCaptureKitSharingSessionManager.mm:
(-[WebDisplayMediaPromptHelper initWithCallback:]):
(-[WebDisplayMediaPromptHelper contentSharingPicker:didCancelForStream:]):
(-[WebDisplayMediaPromptHelper startObservingPicker:]):
(-[WebDisplayMediaPromptHelper stopObservingPicker:]):
(WebCore::ScreenCaptureKitSharingSessionManager::~ScreenCaptureKitSharingSessionManager):
(WebCore::ScreenCaptureKitSharingSessionManager::cancelPicking):
(WebCore::ScreenCaptureKitSharingSessionManager::promptWithSCContentSharingPicker):
(-[WebDisplayMediaPromptHelper contentSharingPickerDidCancel:forStream:]): Deleted.

Canonical link: https://commits.webkit.org/264988@main
